Tdr 487 initial setup

TomJKing opened the pull request:
Draft PR

Initial setup of Grafana on the TDR management account. 

Followed the configuration used for TDR Keycloak

Question whether should use the Jenkins VPC instead of creating a new one.

Database still to be added

TomJKing committed:
Initial Grafana setup

Create basic Grafana AWS resources

Single instance of Grafana running in the TDR management account. Access to other TDR environment metrics via assume roles

Follows configuration for other ECS tasks such as Keycloak on the environment.

Database to be added in separate commit.

TomJKing committed:
Add missing line breaks

TomJKing committed:
Add Cloudwatch plugin used in TDR Prototype

MancunianSam commented:
Looks good so far. 
I would say that you can put this into the same VPC as jenkins. Maybe to avoid confusion we could rename the tdr-jenkins-vpc-mgmt to tdr-vpc-mgmt or something similar although we don't have to.
This would mean that you wouldn't need `flowlogs.tf` and `vpc.tf` in the modules directory which should tidy things up a bit.

The other thing is that there's a lot of terraform code in the `modules/grafana` directory. Would it be better to move as much as possible into the `tdr-terraform-modules` repo?
`ecr.tf` can be moved into `tdr-terraform-modules/ecs` in its own file with its own count and `route53.tf` can be replaced with calling the module from `root.tf`
I'm not sure about the other files but if we shuffle as much as possible onto `tdr-terraform-modules` then it should be a bit tidier.

Last small thing is that we're not (or I'm not anyway) using the `data "template_file"` blocks but are using `templatefile("${path.module}/templates/checksum_lambda.json.tpl", { environment = local.environment ...` instead. This is personal preference though so feel free to ignore it.

TomJKing committed:
Remove template file data blocks

TomJKing committed:
Use terraform modules to build ecs

TomJKing committed:
Pass vpc name tag as variable to ecs module

TomJKing committed:
Move to Jenkins VPC

Jenkins VPC already available in the TDR management account.

Adding Grafana to existing VPC means no need to create new VPC with additional cost involved

TomJKing committed:
Add Grafana database

Use postgres as database as to be consistent with other databases across the TDR environments

TomJKing committed:
Remove unnecessary variables and files

MancunianSam commented on terraform/modules/grafana/security.tf:
```diff
@@ -0,0 +1,76 @@
+resource "aws_security_group" "database" {
+  name        = "tdr-grafana-database-security-group-${var.environment}"
+  description = "Allow inbound access from the Grafana load balancer only"
+  vpc_id      = var.vpc_id
```
Should this say "access to and from the ECS task only" or something like that ?

TomJKing replied:
Reworded

MancunianSam commented on terraform/modules/grafana/subnets.tf:
```diff
@@ -0,0 +1,63 @@
+# Bring default security group under Terraform management and remove all rules
+resource "aws_default_security_group" "default" {
+  vpc_id = var.vpc_id
+}
```
Do you need this? There's a resource for the default security group already being managed in the Jenkins terraform and there's only one default security group per VPC.

TomJKing replied:
removed

MancunianSam commented on terraform/root_variables.tf:
```diff
@@ -0,0 +1,31 @@
+  description = "forms the second part of the bucket name, eg. upload"
+  default     = "grafana"
+}
+
```
For the `function` and `project` variables, they both mention bucket names but they're in the root_variables here so it's not clear what they refer to. Maybe the function one could reference the alb. I don't think you need to mention a bucket at all for the `project` variable, just leave it at "abbreviation for the project" What do you think?

TomJKing replied:
Reworded

TomJKing committed:
Pull request feedback
